### PR TITLE
Add option to specify a timezone for events

### PR DESCRIPTION
**Pull request by denispapec**
## Proposed change

New `timezone` option forces the events to use a specific timezone when provided calendar file is missing that information.

Closes #2575 

## Type of change

- [ ] New service widget
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [x] If applicable, I have added corresponding documentation changes.
- [ ] If applicable, I have reviewed the [feature](https://gethomepage.dev/latest/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [x] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/latest/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/latest/more/development/#code-linting).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

File: docs/widgets/services/calendar.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ widget:
       url: https://domain.url/with/link/to.ics # URL with calendar events
       name: My Events # required - name for these calendar events
       color: zinc # optional - defaults to pre-defined color for the service (zinc for ical)
+      timezone: America/Los_Angeles # optional - force timezone for events when they are shown at the wrong time
```
Comment (shamoon) [inline docs/widgets/services/calendar.md:30]: Does this mean, when they dont have a timezone specified? I think better to state that way.

File: src/widgets/calendar/event.jsx [hunk 2]
```diff
@@ -39,3 +39,7 @@ export default function Event({ event, colorVariants, showDate = false, showTime
     </div>
   );
 }
+
+export function compareIfSameAsEventDateTime(date, event) {
```
Comment (shamoon) [inline src/widgets/calendar/event.jsx:43]: more like `compareDateTimezoneAware` or something?

**Review by shamoon — COMMENTED** [review]
It's certainly not your fault but this code has become extremely hard to parse. Im fine basically assuming this works...

File: docs/widgets/services/calendar.md (hunk 1 above)
Comment (denispapec) [reply docs/widgets/services/calendar.md]: Documentation is now updated. If it's the same timezone - no change, if missing or different in ical - will be converted to this timezone

**Comment by denispapec** [comment]
> It's certainly not your fault but this code has become extremely hard to parse. Im fine basically assuming this works...

Yeah, I agree. Seems to be working with replicated issues.

**Review by shamoon — APPROVED** [review]
Cheers thanks as always
